### PR TITLE
DOCS Improve rendering of JavaScript property types

### PR DESCRIPTION
**Pull request by hoodmane**
<details>
<summary>Before:</summary>

Screenshot from 2023-01-05 12-29-55

</details>

<details>
<summary>After</summary>

Screenshot from 2023-01-05 12-30-02

</details>

### CODE REVIEW

**Review by ryanking13 — APPROVED** [review]
Very cool, thanks for the documentation works @hoodmane!

File: docs/sphinx_pyodide/sphinx_pyodide/__init__.py [hunk 1]
```diff
@@ -12,7 +12,38 @@ def wrap_analyzer(app):
...
+    from pathlib import Path
+
+    from jinja2 import ChoiceLoader, Environment, FileSystemLoader, PackageLoader
+    from sphinx_js.analyzer_utils import dotted_path
+    from sphinx_js.renderers import JsRenderer
+
+    loader = ChoiceLoader(
+        [
+            FileSystemLoader(Path(__file__).parent / "templates"),
+            PackageLoader("sphinx_js", "templates"),
```
Comment (ryanking13) [inline docs/sphinx_pyodide/sphinx_pyodide/__init__.py:28]: By the way, it seems like [sphinx-js](https://github.com/mozilla/sphinx-js) is having some new contributors recently. So maybe we can upstream some of our changes.

Comment (hoodmane) [reply docs/sphinx_pyodide/sphinx_pyodide/__init__.py:28]: It seems like we have some significant difficulties in getting up to sphinx-js v3.2 -- sphinx-js pins typedoc version 0.15.0, but v0.15.0 cannot process our source files so we use v0.22.15. We also pin sphinx-js v3.1.0 and these work fine together. However, when you try to use sphinx-js v3.2.1 with typedoc 0.22.15, it crashes.

Hopefully upgrading to spinx-js 3.1.2 will be okay. Seems like we have to rewrite flatten-suffix-tree

Comment (ryanking13) [reply docs/sphinx_pyodide/sphinx_pyodide/__init__.py:28]: Thanks for the observation! So the life was not that easy :(

**Comment by hoodmane** [comment]
Backported to stable 4e84ea80d5aaa296b13f79eba7e28313d0a90eb2

**Pull request merged**